### ISSUE TITLE
Flutter app UI

### ISSUE THREAD

**Issue by RichPerspective007** [Shinjan-saha/Canteen-Website#17]
I want to work on the UI of the Flutter Canteen App. Currently, I installed the app and there was just a blank white page. 

I will first work on UI of the app. I can work with Firebase to set up user authorization. I am not proficient in state management, but I am learning Riverpod from YouTube.

Please assign me and if you have any questions or want to discuss about the app, I am willing to answer.

**Comment by Shinjan-saha** [comment]
App will have the same content as the Website it has an Webview Package your mobile must be get connected to internet for it to work

**Comment by RichPerspective007** [comment]
https://github.com/Shinjan-saha/Canteen-Website/assets/121783925/5a010d3d-e576-478a-abe6-d344a976c879
This is a screen recording of the working of the app in my mobile device. Though the network is really strong, nothing other than a white page is visible.

**Comment by Shinjan-saha** [comment]
Yeah realized the issue the problem is that network traffic is not configured & we need a network security config file it's not configured for this project . that is why in web view package it's not rendering the website link:- https://canteen-website-shinjan.vercel.app/
<img width="733" alt="network-issue" src="https://github.com/Shinjan-saha/Canteen-Website/assets/116014977/6fdd14f2-9061-40f0-b106-2cf0c6823e6c">
Here is Terminal where you can see the network option not permitted & failure in creating a http server bind 

can you solve this issue? it's not an UI Problem it's an network security problem

**Comment by RichPerspective007** [comment]
I don't know the exact solution, but it's pretty clear that there are some issues with permissions and I can look into that. I am confident that I can find the solution. Can I work on this issue?

**Comment by Shinjan-saha** [comment]
Okay I am Assigning you this issue

**Comment by Shinjan-saha** [comment]
Update?

**Comment by RichPerspective007** [comment]
I am really sorry for the immense delay I have caused. Actually I had started working on the problem right away and when I cloned the project on my device and ran the app in the emulator and also using my actual mobile phone for debugging, in both cases the app works just fine. The debug console does not show the errors related to network permissions anymore.

I did not create an apk to test the app on the device by manual installation. I wanted to consult with you first but I unexpectedly came down with severe case of stomach problems due to my bad eating habits off late. 

I have attached screenshots of the app running well in debug mode in my phone and screenshot from the debug console.
![m1](https://github.com/Shinjan-saha/Canteen-Website/assets/121783925/78ece4fe-65d6-4e4c-a705-684044d74f60)
![m2](https://github.com/Shinjan-saha/Canteen-Website/assets/121783925/014aeed0-e8b6-467e-87c4-7a9790ca4099)
![debcan](https://github.com/Shinjan-saha/Canteen-Website/assets/121783925/361779c7-8b65-46bb-8040-3a15a5b5779f)

**Comment by Shinjan-saha** [comment]
Yeah the App will work in debug mode but it creates the problem in it's release version. So the network issue must be solved if we want the release version  to work fine

**Comment by RichPerspective007** [comment]
I have implemented a solution. The app runs in the release mode when I use the command 
`flutter run --release`

I just took internet permissions for iOS and android. But since I do not have iOS simulators, I am not able to confirm whether rhe solution works for iOS or not.

So, should I create a PR? Also, if you want me to explain in detail, I will do that gladly.

**Comment by Shinjan-saha** [comment]
What is the solution that you have implmented?

**Comment by RichPerspective007** [comment]
In the AndroidManifest.xml file, I added permissions to access content from the internet. 

`<uses-permission android:name="android.permission.INTERNET" />`
The xml file is located in android/app/src/main/AndroidManifest.xml

I did not know how to implement the same for iOS so I got the solution from stack overflow for iOS. 
There the same solution was given for android as well. 
For iOS,

in ios/Runner/Info.plist, I added:

`       <key>io.flutter.embedded_views_preview</key>
	<string>YES</string>      `
	
under the dict tag. From what I understand, it is a key value pair of permissions for the iOS app


After doing this, I ran the command
`flutter clean`

and then again set up the project using:
`flutter pub get`

and then I ran the app in the emulator and in my Android phone using USB debugging in release mode using the command:
`flutter run --release`

Since it ran in release mode and the webview worked, I assume that the solution will work well in release builds apks. 

However, I am somewhat skeptical of the iOS solution because just string YES seems like a very hardcoded way of giving permissions.

**Comment by RichPerspective007** [comment]
Is this an acceptable solution? Can I create a PR?

**Comment by Shinjan-saha** [comment]
Is it solving the white screen problem on the build apk ?
